tests/dflowfm/test_mdu.py

```diff
@@ -84,10 +89,68 @@ def test_mdu_with_optional_sections(self):
         assert fm_model.veg.stemheightstd == 0.0
 
     def test_disk_only_file_model_list_fields_are_initialized_correctly(self):
-        data = {"crsfile": [Path("test.crs")]}
+        data = {"landboundaryfile": [Path("test.ldb")]}
 
-        model = Output(**data)
+        model = Geometry(**data)
 
-        assert model.crsfile is not None
-        assert len(model.crsfile) == 1
-        assert isinstance(model.crsfile[0], DiskOnlyFileModel)
+        assert model.landboundaryfile is not None
+        assert len(model.landboundaryfile) == 1
+        assert isinstance(model.landboundaryfile[0], DiskOnlyFileModel)
+
+
+class TestOutput:
+    """Test class to test the [output] section in an MDU file."""
+
+    def test_obs_crs_from_mdu(self):
+        fmmodel = FMModel(
+            filepath=test_input_dir
+            / "e02"
+            / "f101_1D-boundaries"
+            / "c01_steady-state-flow"
+            / "Boundary.mdu"
+        )
+
+        assert isinstance(fmmodel.output.crsfile[0], ObservationCrossSectionModel)
+
+        # Check recursive construction of ObservationCrossSectionModel object
+        assert len(fmmodel.output.crsfile[0].observationcrosssection) == 4
+        assert (
+            fmmodel.output.crsfile[0].observationcrosssection[0].name
+            == "ObservCross_Chg_T1"
+        )
+        assert (
+            len(fmmodel.output.crsfile[0].observationcrosssection[3].xcoordinates) == 2
+        )
+
+    def test_mixed_obs_crs_files(self):
```

Comment:
We have seen that this test gives a different result compared to actually reading a `.mdu` file where you list both the `test.pli` and `ObservationPoints_crs.ini`.

I think writing a test that does the above has several advantages:
- You test the package the way the user would use it.
- It is more resistant to refactorings. Assume we add an additional argument to the `Parser` in the source code. You would have to update this test as well. While if you simply read the mdu file, you do not have to change this test. In both cases (reading a `.mdu` and the test you suggest) test if we successfully create a `PolyFile` for the `.pli` and an `ObservationCrossSectionModel` from `ObservationPoints_crs.ini`
- It makes the test more readable (easier to follow what happens). Of course, this is subjective :)

Reply:
If it was your intention to test the `Parser` instead, I would have checked if the given input would return the correct `Document` instead.

Reply:
I agree with you: I have removed this parser-based test and replaced it by three FMModel loading tests, similar to what you have done for the obs ini/xyn mixed testing. See newer commits.